util/util.go

```diff
@@ -152,3 +154,17 @@ func AddImageNames(store storage.Store, image *storage.Image, addNames []string)
 	}
 	return nil
 }
+
+// GetFailureCause checks the type of the error "err" and returns a new
+// error message that reflects the reason of the failure.
+// In case err type is not a familiar one the error "defaultError" is returned.
+func GetFailureCause(err, defaultError error) error {
+	switch nErr := errors.Cause(err).(type) {
+	case errcode.Error:
+		return nErr
+	case *url.Error:
+		return nErr
```

Comment:
Is it intentional to return nErr in both cases?

Reply:
Yes, I tried to use `falltrhough` but I got compiled error that said I can't do it.
Maybe I can change it a little bit so it will be more readable.

Reply:
This should work.
```
case errcode.Error, *url.Error:
      return nErr
```